soroban-sdk/src/env.rs

```diff
@@ -442,6 +442,19 @@ impl Env {
     pub fn log_value<V: IntoVal<Env, RawVal>>(&self, v: V) {
         internal::Env::log_value(self, v.into_val(self)).unwrap_infallible();
     }
+
+    /// Replaces the executable of the current contract with the provided Wasm.
+    /// 
+    /// The Wasm blob identified by the `wasm_hash` has to be already present
+    /// on-chain (the upload happens via `INSTALL_CONTRACT_CODE` host function
+    /// or via `install_contract_wasm` test function in unit tests).
+    /// 
+    /// The function won't do anything immediately. The contract executable
+    /// will only be updated after the invocation has successfully finished.
+    pub fn update_current_contract_wasm(&self, wasm_hash: &BytesN<32>) {
```

Comment:
Could we put this function into the deployer type? This function adds a way to deploy code to an existing contract. i.e. add this function to `DeployerWithCurrentContract` as `update`?

Reply:
Isn't that a bit confusing? Deployer deploys another contract, while update is related to the current one... Maybe `update_current_wasm` then?

Reply:
I agree using the deployer is a little confusing. If you do make this change though, wouldn't it make more sense to put it under `Deployer` instead of `DeployerWithCurrentContract` and `DeployerWithOtherContract`?

Reply:
I think you're right. I mostly would like to avoid the `Env` becoming a "god-object" where the scope/impact of one function compared to another and their relationships become blurred. This is something we had at one point until we categorized functionality and started moving fns out under sub-functions. It feels like this function belongs somewhere else, but `Deployer` might not be the best place.

We can move it later and do not need to block on this.

Reply:
Sounds good, we still have time to reorganize this somehow if needed.